source/fuzz/transformation_propagate_instruction_down.cpp

```diff
@@ -0,0 +1,589 @@
+// Copyright (c) 2020 Vasyl Teliman
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_propagate_instruction_down.h"
+
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+
+namespace spvtools {
+namespace fuzz {
+
+TransformationPropagateInstructionDown::TransformationPropagateInstructionDown(
+    const protobufs::TransformationPropagateInstructionDown& message)
+    : message_(message) {}
+
+TransformationPropagateInstructionDown::TransformationPropagateInstructionDown(
+    uint32_t block_id, uint32_t phi_fresh_id,
+    const std::map<uint32_t, uint32_t>& successor_id_to_fresh_id) {
+  message_.set_block_id(block_id);
+  message_.set_phi_fresh_id(phi_fresh_id);
+  *message_.mutable_successor_id_to_fresh_id() =
+      fuzzerutil::MapToRepeatedUInt32Pair(successor_id_to_fresh_id);
+}
+
+bool TransformationPropagateInstructionDown::IsApplicable(
+    opt::IRContext* ir_context,
+    const TransformationContext& transformation_context) const {
+  // Check that we can apply this transformation to the |block_id|.
+  if (!IsApplicableToBlock(ir_context, message_.block_id())) {
+    return false;
+  }
+
+  const auto successor_id_to_fresh_id =
+      fuzzerutil::RepeatedUInt32PairToMap(message_.successor_id_to_fresh_id());
+
+  for (auto id : GetAcceptableSuccessors(ir_context, message_.block_id())) {
+    // Each successor must have a fresh id in the |successor_id_to_fresh_id|
+    // map. We use overflow ids if available.
```

Comment:
"map.  We use overflow ids ..." -> "map, unless overflow ids are available."

Reply:
Fixed in b6fe77b.